robot_skills/src/robot_skills/arms.py

```diff
@@ -298,21 +298,21 @@ def __init__(self, robot_name, tf_listener, side):
         """
         super(Arm, self).__init__(robot_name=robot_name, tf_listener=tf_listener)
         self.side = side
-        if (self.side is "left") or (self.side is "right"):
-            pass
-        else:
-            raise Exception("Side should be either: left or right")
 
         self._occupied_by = None
 
         self._operational = True  # In simulation, there will be no hardware cb
 
         # Get stuff from the parameter server
-        offset = self.load_param('skills/arm/offset/' + self.side)
+        offset = self.load_param('skills/arm/' + self.side + '/grasp_offset/')
         self.offset = kdl.Frame(kdl.Rotation.RPY(offset["roll"], offset["pitch"], offset["yaw"]),
                                 kdl.Vector(offset["x"], offset["y"], offset["z"]))
 
-        self.marker_to_grippoint_offset = self.load_param('skills/arm/offset/marker_to_grippoint')
+        self.marker_to_grippoint_offset = self.load_param('skills/arm/' + self.side + '/marker_to_grippoint')
+
+        # Grasp offsets
+        go = self.load_param('skills/arm/' + self.side + '/base_offset')
+        self._base_offset = kdl.Vector(go.get("x"), go.get("y"), go.get("z"))
```

Comment:
Why the `.get`? If it's a dictionary, you can also do the `['x']` or `.get('x', default_val)`. If a `kdl.Vector` needs x,y,z and the dict provides those three, you could even do `kdl.Vector(**go)`.